BUILD.md

```diff
@@ -0,0 +1,86 @@
+>> [`./releng/docker/README.md`](https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/docker/README.md) More detailed notes about the containerised build process and how to manipulate it.  
+
+## Dependencies
+
+To build itself, ZFSBootMenu relies on either [`podman`](https://podman.io) or [`docker`](https://www.docker.com), preferring the former.
+
+ZFSBootMenu can be flexible regarding which container management tool is used; they are, to an extent, interchangeable.
+
+>If a custom build environment creation is desired, [`buildah`](https://buildah.io) will used in tandem with `podman` to do so. Further information can be found [above](https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/docker/README.md).  
+
+It may be easier to set up one over the other, depending on which flavour of Linux is in use.
+
+> ### Podman  
+>  
+> According to your distribution's particular package manager, install `podman`.  
+> `# pacman -S podman`  
+> `# apt-get install podman`  
+> `...`  
+
+Information on setting up `podman` to allow rootless container deployment can be found [here](https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md).
+
+> ### Docker  
+>  
+> According to your distribution's particular package manager, install `docker`...  
+>  
+> `# pacman -S docker`  
+> `# apt-get install docker`  
+> `...`  
+>  
+> ...and `-a`ppend any desired usernames to the `docker` user`-G`roup, to allow these users to work with images and containers *sans root.*  
+>  
+> `# usermod -a -G docker <jrtoddy>`  
+
+## Obtaining the Sources
+
+The Git `clone` command will create a copy of the current ZFSBootMenu source hierarchy in a chosen target `<directory>`; by default, this will be `./zfsbootmenu` if no other argument is given.
+
+> `$ git clone https://github.com/zbm-dev/zfsbootmenu.git <directory>`  
+
+Likewise, a tarball can be obtained and extracted using normal command line tools.
+
+> `$ wget https://github.com/zbm-dev/zfsbootmenu/archive/master.tar.gz`  
+> `$ tar -xvf master.tar.gz`  
+
+## The Process
+
+The ZFSBootMenu build process is initiated by the `zbm-builder.sh` script in the root of the repository.
+
+It assumes that the local system is the target, and, as such, it attempts to fetch a copy of the local [`/etc/zfs/zpool.cache`](https://openzfs.github.io/openzfs-docs/Project%20and%20Community/FAQ.html#the-etc-zfs-zpool-cache-file) and [`/etc/hostid`](https://openzfs.github.io/openzfs-docs/man/8/zgenhostid.8.html) to be integrated into the `initramfs`.
+
+This behaviour can be disabled by invoking `zbm-builder.sh` with the `-C` and `-H` arguments, to disable use of the local `C`ache and `H`ostid respectively.
+
+It will then launch a temporary container via [`docker`](http://www.docker.com) or [`podman`](https://www.podman.io) using the image `ghcr.io/zbm-dev/zbm-builder:latest`. 
```

Comment:
Something like `It will then launch an ephemeral container via your chosen container engine using the default image ...` would avoid repeating podman/docker again.

Reply:
:+1: